### DISCUSSION TITLE
[CIP-00xx] Standard for Injected web3 Provider

### DISCUSSION THREAD

**Discussion by daoauth** [celo-org/celo-proposals#329] [CIP Discussions]
## Overview
- Standard to find out the type of web3 provider to support dapps running in web browsers

## Goals
- In the case of web DApps, you can only use [DSRV Celo Extension Wallet](https://chrome.google.com/webstore/detail/celo-desktop-wallet/kkilomkmpmkbdnfelcpgckmpcaemjcdh) or MetaMask Extension wallet, but a standard is required to distinguish them.
- Through these standards, each provider can be identified and transactions can be sent to the Celo blockchain through appropriate operations.

## Proposed Solution
- Ready `ethereum.isMetaMask` property for MetaMask. Through this property, it can be seen that it is a provider injected in MetaMask.
- So, it is suggested that providers injected in Celo have a property with the syntax of `celo[providerName]`.
  * eg. Some injected provider has `celo.isDesktop` property. It's injected by DSRV Celo Extension wallet
  * eg. For DApp browser has `celo.isMobile` too

## Useful Links
- [MetaMask](https://docs.metamask.io/guide/ethereum-provider.html#properties)

**Comment by prestwich** [comment]
Associated CIP PR: 
https://github.com/celo-org/celo-proposals/pull/57/files
